src/parser/core/modules/Swiftcast.tsx

```diff
@@ -47,6 +47,25 @@ export abstract class SwiftcastModule extends BuffWindowModule {
 		severityTiers: this.severityTiers,
 	}
 
+	protected init() {
+		super.init()
+		// Inheriting the class doesn't update expectedGCDs's parameters when they
+		// override, so let's (re)define it here... feels mega jank tho
+		this.expectedGCDs = {
+			expectedPerWindow: 1,
+			suggestionContent: this.suggestionContent,
+			severityTiers: this.severityTiers,
+		}
+	}
+
+	// Provide our own logic for the end of the fight – even though the window is
+	// ~4 GCDs 'wide', we can only use one action with it anyway; this change should
+	// ding them only if they had enough time during the window to use a spell with
+	// swiftcast
+	protected reduceExpectedGCDsEndOfFight(buffWindow: BuffWindowState): number {
+		return Math.min(1, super.reduceExpectedGCDsEndOfFight(buffWindow))
```

Comment:
Contra your comment, what this implementation will do is if they use SC with less than 10 seconds before the end of the fight, it will reduce the expected GCDs to 0 (because super is going to return `Math.ceil((windowDurationMillis - fightTimeRemaining)/this.globalCooldown.getEstimate)`, or 10000 - 9500 / 2500 rounded up to 1 if they use it 9.5s before the end of the fight with a 2.5 GCD.

If you want it to treat using with 2.5s+ left in fight as expect 1, and otherwise expect 0, you'd need to implement something like:
```suggestion
		if ( this.buffStatus.duration ) {
			// Check to see if this window is rushing due to end of fight - reduce expected GCDs accordingly
			const fightTimeRemaining = this.parser.fight.end_time - buffWindow.start
			const gcdEstimate = this.globalCooldown.getEstimate()

			return ( fightTimeRemaining > gcdEstimate ) ? 0 : 1
		}

		return 0
```

Reply:
That's a good point, and after checking back in the code I do understand it a bit better. I'll push another commit with the changed code.